src/oss-find-squats-lib/MutateExtension.cs

```diff
@@ -188,5 +190,23 @@ public static async IAsyncEnumerable<FindPackageSquatResult> EnumerateExistingSq
 
             }
         }
+
+        /// <summary>
+        /// Adds a <see cref="Mutation"/> to the dictionary of generated mutations.
+        /// </summary>
+        /// <param name="generatedMutations">The dictionary of generated mutations to add <paramref name="mutation"/> to.</param>
+        /// <param name="mutation">The <see cref="Mutation"/> to add to the dictionary.</param>
+        private static void AddMutation(this IDictionary<string, IList<Mutation>> generatedMutations, Mutation mutation)
```

Comment:
nit: This might be a little cleaner with `AddOrUpdate` (requires ConcurrentDictionary) or `GetValueRefOrAdd`

for the second:
```c#
ref MyType value = CollectionsMarshal.TryGetValueRef(generatedMutations, mutationString, out bool exists);

if (exists)
{
    value.Add(mutation);
}
else
{
    ref value = new List<Mutation>() { mutation };
}

```



https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-6.0
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.collectionsmarshal.getvaluereforadddefault?view=net-6.0
https://github.com/dotnet/runtime/issues/27062